iOS/Models/SettingsTableSource.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Esri.ArcGISRuntime.OpenSourceApps.IndoorRouting.iOS.Helpers;
+using Foundation;
+using UIKit;
+
+namespace Esri.ArcGISRuntime.OpenSourceApps.IndoorRouting.iOS.Models
+{
+    internal class SettingsTableSource : UITableViewSource
+    {
+        // Cell reuse identifiers
+        private const string HomeLocationCellIdentifier = nameof(HomeLocationCellIdentifier);
+        private const string LocationServicesCellIdentifier = nameof(LocationServicesCellIdentifier);
+        private const string RoutingCellIdentifier = nameof(RoutingCellIdentifier);
+        private const string UseOnlineBasemapCellIdentifier = nameof(UseOnlineBasemapCellIdentifier);
+
+        private readonly MapViewModel _viewModel;
+
+        private UISwitch _locationSwitch;
+        private UISwitch _routingSwitch;
+        private UISwitch _useOnlineBasemapSwitch;
+
+        public SettingsTableSource(MapViewModel viewModel) => _viewModel = viewModel;
+
+        /// <summary>
+        /// Returns the setting cell for a particular row
+        /// </summary>
+        public override UITableViewCell GetCell(UITableView tableView, NSIndexPath indexPath)
+        {
+            switch (indexPath.Row)
+            {
+                case 0:
```

Comment:
I would consider adding comments describing the cases. We dont have context to what they are here.